### ISSUE TITLE
File formats section needs to be filled out

### ISSUE THREAD

**Issue by DrDap** [qsimulate/QSP_Materials_Documenation#12]
There is a file formats section of the manual that has not been developed. We should provide the formats for any files that are either imported at the user level or that the user can download.

**Comment by spheex** [comment]
I filled  this page with some information about:

- input file SMILES and XYZ: I do not know if another file format works or will work? or if multi-frame XYZ files are supported?
- output structure: described it as XYZ
- output data: CSV file with properties, etc. --> are the chemical elements present in the "dft_geom" column (list of coordinates)? If not, is the order of atoms the same as in the original SMILES/XYZ (and what about hydrogens)?

**Comment by DrDap** [comment]
@Samthos should answer these definitively:

- What SMILES formats for input do we support? At one time it had to be [SMILES1,SMILES2,...]. Do we still require that, or can we accept one SMILES and name per line (the way most of the world deals with SMILES).

- Also answer re: CSV output

It looks like we're not supporting input/output for restarts at Workspace/Project/Batch level in this interaction of the platform, so I guess we don't have to document those.

**Comment by Samthos** [comment]
Format for smiles is a effectively JSON list.  Each element in the list must provide a smiles and optionally can provide a name for the molecule (also called smiles tag). The smiles and name are provided in the same string and are separated by a space. Everything to the right of the space becomes the name. If no name is provide, name takes the value of the smiles string. 

For example
```
["C", "CC ethane", 
"CCO  johnny walker"]
```

(please double check that input) would give
```
[
    {"smiles": "C", "name": "C"},
    {"smiles": "CC", "name": "ethane",
    {"smiles": "CCO", "name": "johnny walker"}
]
```

**Comment by Samthos** [comment]
CSV: Elements are recorded by their atomic number. Hydrogens are included in the record.  

Order is not the same as the original file. The atoms are ordered atomic charge descending.

**Comment by spheex** [comment]
Using the tag in the json file works! @Samthos 
![image](https://user-images.githubusercontent.com/70961129/158786446-f940a247-f012-4843-b79a-956bfb27bf12.png)

**Comment by spheex** [comment]
@Samthos Question:
To what corresponds the "smiles_id" in the output ,csv?
For the example above, the "smiles_id" of "C" is 2, "CC" is 4 and "CCO" is 6.
I cannot figured out to what it is related?

**Comment by Samthos** [comment]
smiles id is a part of a set of indices that give a molecule a unique identifier, and it doesn't use outside of the platform. 

It goes 2->4-6 because the smiles strings used for calculation aren't the ones uploaded by the client. The actual process is 
```
for orginal_SMILE in uploaded_SMILEs:
    insert original_smile into db
    canonical_SMILEs = canonicalize(original_SMILE) # this is a list
    for cs in canonical_SMILEs:
       insert cx into db (parent == original_SMILE
```
The canonicalize function returns a list because stereocenters must be explicitly on the backend. If a smiles is uploaded with the "all_stereoisomer" (may be called all permutation) option enable, all unassigned stereo centers are assigned. If the option is disabled, all stereocenters must be assigned at upload.

See the two [ccno molecules](https://github.com/qsimulate/MI_frontend/tree/main/tests/src/auto_tester/resources/molecules) for an example of the stereoisomers.

**Comment by spheex** [comment]
Thank you. So if I understand correctly, the smiles_id in the .csv file will never be useful for the users?

**Comment by Samthos** [comment]
1. No nproc or mem lines. We support the connectivity block after the geometry block. 
2. xyz or gjf with a specific format for the comment lines. I'll DM you some examples
3. standard cif

**Comment by spheex** [comment]
According to this documentation: https://www.iucr.org/__data/assets/pdf_file/0019/22618/cifguide.pdf

A CIF may contain one or more sets of data. Each data set is preceded by a code beginning with the characters data_.

Do we support that in the platform? @Samthos

**Comment by Samthos** [comment]
No, but more practical reasons.  All of the workflows that need cif upload only support uploading a single structure.
